### ISSUE TITLE
Trailing whitespace for strings to be translated

### ISSUE THREAD

**Issue by vorlif** [wagtail/wagtail#9596]
### Issue Summary

More a minor cosmetic flaw than a bug, but bug still fit best.

I noticed that there are strings to be translated where a space is inserted at the end to keep the distance to the following text. However, the spaces are by default not highlighted in transifex and therefore many translators do not add them. As a result, the spacing between the text parts is missing in many languages.

Example: German

![image](https://user-images.githubusercontent.com/4333770/199825010-6f809011-cf69-4786-81db-a1fb73d7379e.png)

I found these two places where a space was added:

1. [/wagtail/admin/templates/wagtailadmin/shared/headers/page_create_header.html#L5](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/headers/page_create_header.html#L5)
2. [/wagtail/snippets/templates/wagtailsnippets/snippets/headers/create_header.html#L15](https://github.com/wagtail/wagtail/blob/main/wagtail/snippets/templates/wagtailsnippets/snippets/headers/create_header.html#L15)

Possible solutions:

1. We add the space character in the code/template ([like here, `|add:' '`](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/shared/side_panels/includes/status/workflow.html#L104)). This assumes, for each language, that there is a space between the colon (or a character of the respective language) and the following text.
2. We use `blocktranslate` and add the following text as a variable. ([like here](https://github.com/wagtail/wagtail/blob/main/wagtail/snippets/templates/wagtailsnippets/snippets/headers/edit_header.html#L17))
3. (We add a comment for the translators that they should respect the space.)

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Create a snippet model
3. Change your profile language to German
4. Create a new snippet
5. Look at the text in the header

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.10.8
-   Django version: 4.1.1
-   Wagtail version: 4.1
-   Browser version: Not relevant

**Comment by vorlif** [comment]
I would also make the changes myself, just want to know beforehand if they are wanted and which solution fits best.

**Comment by allcaps** [comment]
I'd vote for `blocktrans`
```
    {% trans 'New: '|add:content_type.model_class.get_verbose_name as title %}
```
->
```
{% with title=content_type.model_class.get_verbose_name %}
    {% blocktrans trimmed %}New: {{ title }}{% endblocktrans %}
{% endwith %}
```

**Comment by lb-** [comment]
Yes. We should use blocktrans for this kind of thing with the trimmed option.

**Comment by vorlif** [comment]
@allcaps @lb- Thanks for your feedback.

I created a PR to remove the two points that caught my eye. In the po files I found more strings with trailed whitespace. However, I didn't notice in the admin interface that this is a problem anywhere. In some languages the spaces were appended as well. I would leave it that way for now because of this, and if I notice more places in the frontend, I open a new PR.
